src/components/Geolocation.tsx

```diff
@@ -0,0 +1,124 @@
+   */
+  geolocationOptions?: PositionOptions,
+  /**
+   * The radius, in miles, around the user's location to find results. Defaults to 50.
+   * If location accuracy is low, a larger radius may be used automatically.
+   */
+  radius?: number,
+  /** The label for the button. Defaults to 'Use my location'. */
+  label?: string,
+  /** Custom icon component to display along with the button. */
+  GeolocationIcon?: React.FC,
+  /**
+   * A function which is called when the geolocation button is clicked.
+   * This is called prior to executing the existing click behavior.
+   */
+  onClick?: () => void,
+  /** CSS classes for customizing the component styling. */
+  customCssClasses?: GeolocationCssClasses
+}
+
+const LOCATION_FIELD_ID = 'builtin.location';
+const METERS_PER_MILE = 1609.344;
+
+/**
+ * A React Component which collects location information to create a
+ * location filter and perform a new search.
+ *
+ * @public
+ *
+ * @param props - {@link GeolocationProps}
+ * @returns A react component for geolocation
+ */
+export function Geolocation({
+  geolocationOptions,
+  radius = 50,
+  label = 'Use my location',
+  //TODO: replace default icon with SVG create from design team
+  GeolocationIcon = YextIcon,
+  onClick,
+  customCssClasses,
+}: GeolocationProps): JSX.Element | null {
+  const searchActions = useSearchActions();
+  const staticFilters = useSearchState(s => s.filters.static || []);
+  const [isFetchingLocation, setIsFetchingLocation] = useState<boolean>(false);
+  const cssClasses = useComposedCssClasses(builtInCssClasses, customCssClasses);
+
+  const handleGeolocationClick = useCallback(async () => {
+    onClick?.();
+    setIsFetchingLocation(true);
+    try {
+      const position = await getUserLocation(geolocationOptions);
+      const { latitude, longitude, accuracy } = position.coords;
+      const locationFilter: SelectableStaticFilter = {
+        displayName: 'Current Location',
+        selected: true,
+        filter: {
+          kind: 'fieldValue',
+          fieldId: LOCATION_FIELD_ID,
+          matcher: Matcher.Near,
+          value: {
+            lat: latitude,
+            lng: longitude,
+            radius: Math.max(accuracy, radius * METERS_PER_MILE)
+          },
+        }
+      };
+      const nonLocationFilters = staticFilters?.filter(filter => {
+        return !(filter.filter.kind === 'fieldValue'
+          && filter.filter.fieldId === LOCATION_FIELD_ID);
+      }) ?? [];
+      searchActions.setStaticFilters([...nonLocationFilters, locationFilter]);
+      executeSearch(searchActions);
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      setIsFetchingLocation(false);
+    }
+  }, [geolocationOptions, onClick, radius, searchActions, staticFilters]);
+
+  return (
+    <div className={cssClasses.geolocationContainer}>
+      <button className={cssClasses.button} onClick={handleGeolocationClick}>
+        {label}
```

Comment:
Should this button still be clickable after we've gotten the user's location?

Reply:
I think it's okay for user to still click it and update their location after if it potentially change somehow

Reply:
They could move since the last search

Reply:
I was thinking about mirroring what LocationBias currently does but that makes sense

Reply:
LocationBias still lets you click on the button after we've gotten the user's location

Reply:
Ohh the behavior I was remember was that if the user clicks "block" when the location request prompt pops up the button will disappear. It looks like this only happens in the SDK's LocationBias component and not the react one, which is fine.